set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java

```diff
@@ -114,11 +114,126 @@ public void create_UserSet_400_unauthorized_InvalidJWTToken() throws Exception {
 		.andExpect(status().is(HttpStatus.UNAUTHORIZED.value()));
     }
 
+    // create Entity user set validation tests
+	@Test
+	public void create_EntityUserSet_Unauthorized_InvalidUserRole() throws Exception {
+		String requestJson = getJsonStringInput(ENTITY_USER_SET_REGULAR);
+
+		mockMvc.perform(post(BASE_URL).queryParam(CommonApiConstants.QUERY_PARAM_PROFILE, LdProfiles.MINIMAL.name())
+				.content(requestJson).header(HttpHeaders.AUTHORIZATION, regularUserToken)
+				.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+				.andExpect(status().is(HttpStatus.UNAUTHORIZED.value()));
+	}
+
+	@Test
+	public void create_EntityUserSet_Unauthorized_EmptyToken() throws Exception {
+		String requestJson = getJsonStringInput(ENTITY_USER_SET_REGULAR);
+
+		mockMvc.perform(post(BASE_URL).queryParam(CommonApiConstants.QUERY_PARAM_PROFILE, LdProfiles.MINIMAL.name())
+				.content(requestJson).header(HttpHeaders.AUTHORIZATION, "")
+				.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+				.andExpect(status().is(HttpStatus.UNAUTHORIZED.value()));
+	}
+
+	//create entity user set with editor token
+	@Test
+	public void create_EntityUserSet_Success() throws Exception {
+		String requestJson = getJsonStringInput(ENTITY_USER_SET_REGULAR);
+
+		String result = mockMvc.perform(post(BASE_URL).queryParam(CommonApiConstants.QUERY_PARAM_PROFILE, LdProfiles.MINIMAL.name())
+				.content(requestJson).header(HttpHeaders.AUTHORIZATION, editorUserToken)
+				.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+				.andExpect(status().is(HttpStatus.CREATED.value())).andReturn().getResponse().getContentAsString();
+
+		String identifier = getSetIdentifier(result);
+		assertNotNull(getSetIdentifier(result));
+		String creator = getSetCreator(result);
+		assertNotNull(creator);
+        assertTrue(StringUtils.contains(creator, getConfiguration().getEntityUserSetUserId()));
```

Comment:
should also check the presence of the user in the contributor list

Reply:
done